streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryTimeOrderedKeyValueBuffer.java

```diff
@@ -74,12 +74,12 @@ public Builder(final String storeName) {
 
         @Override
         public StoreBuilder<StateStore> withCachingEnabled() {
-            throw new UnsupportedOperationException();
+            return this;
```

Comment:
Should we log a warning here and below?

Reply:
I'm not sure I agree with a warning.

I think it's fine to just ignore the call, since the buffer will behave correctly whether you call these methods or not. In fact, I originally had them no-op, and changed it to throw upon review.

There's no publicly facing API to register a buffer, so there's no way that a user could call these methods right now. In fact, the buffer itself is an `internal` class. Therefore, the only way these methods get called are via Streams itself.

My thought is that it would be confusing to see warnings that actually indicate nothing is wrong and about which you can do nothing.

Would you be satisfied with a javadoc explaining the decision?